backend/app/services/search.py

```diff
@@ -246,6 +246,16 @@ async def remove_duplicates(self, all_results):
                 seen.add(tuple(sorted(flat_doc.items())))
         return filtered_results
 
+    def get_unique_values(self, all_values):
+        seen = set()
+        values = []
+        for val in all_values:
+            lower_value = str(val).lower()
+            if lower_value not in seen:
+                seen.add(lower_value)
+                values.append(val)
+        return values
```

Comment:
or ...

```python
def get_unique_values(self, all_values):
    return list({str(v).lower() for v in all_values})
```

The whole point of a set is that just putting the values into a set does what your function is doing!

Reply:
I wanted to preserve the original casing of the first occurring item , so I came up with this approach